src/cargo/ops/cargo_add/mod.rs

```diff
@@ -559,6 +634,21 @@ fn get_latest_dependency(
     }
 }
 
+fn parse_rust_version(rust_version: impl AsRef<str>) -> (u64, u64, Option<u64>) {
+    // HACK: `rust-version` is a subset of the `VersionReq` syntax that only ever has one comparator
+    // with a required minor and optional patch, and uses no other features. If in the future this
+    // syntax is expanded, this code will need to be updated.
+
+    let version = semver::VersionReq::parse(rust_version.as_ref()).unwrap();
+    assert!(version.comparators.len() == 1);
+
+    let version = &version.comparators[0];
+    assert_eq!(version.op, semver::Op::Caret);
+    assert_eq!(version.pre, semver::Prerelease::EMPTY);
+
+    (version.major, version.minor.unwrap(), version.patch)
+}
+
```

Comment:
This isn't a particularly complicated function (and should probably just be moved inside of `get_latest_dependency`), but it did occur to me that I might end up needing a util like it for subsequent `-Zmsrv-policy` features. I'll probably end up making some sort of wrapper struct later that contains convenience methods for dealing with its quirkiness.